_examples/how-to-use-middy-validation-in-your-serverless-app.md

```diff
@@ -0,0 +1,308 @@
+
+```json
+{
+  "name": "middy-validation",
+  "region": "us-east-1",
+  "main": "stacks/index.js"
+}
+```
+
+## Project layout
+
+An SST app is made up of a couple of parts.
+
+1. `stacks/` — App Infrastructure
+
+   The code that describes the infrastructure of your serverless app is placed in the `stacks/` directory of your project. SST uses [AWS CDK]({% link _chapters/what-is-aws-cdk.md %}), to create the infrastructure.
+
+2. `src/` — App Code
+
+   The code that's run when your API is invoked is placed in the `src/` directory of your project.
+
+## Create our infrastructure
+
+Our app is made up of a simple API. The API will read two variables from the request and returns a greeting message as a response
+
+### Creating our API
+
+Now let's add the API.
+
+{%change%} Replace the `routes` object in `stacks/MyStack.js`.
+
+```js
+routes: {
+    "GET /": "src/lambda.handler",
+}
+```
+
+With the below,
+
+```js
+routes: {
+    "POST /": "src/lambda.handler",
+}
+```
+
+We are using the SST [`Api`](https://docs.serverless-stack.com/constructs/Api) construct to create our API. It simply has one endpoint (the root). When we make a `POST` request to this endpoint the Lambda function called `handler` in `src/lambda.js` will get invoked.
+
+{%change%} Replace the code in `src/lambda.js` with below.
+
+```js
+export async function handler(event) {
+  const { fname, lname } = event.body;
+  return {
+    statusCode: 200,
+    headers: { "Content-Type": "text/plain" },
+    body: `Hello, ${fname + "-" + lname}.`,
+  };
+}
+```
+
+We are reading two variables `fname` and `lname` from the event body and returning a simple greeting message
+
+And let's test what we have so far.
+
+## Starting your dev environment
+
+{%change%} SST features a [Live Lambda Development](https://docs.serverless-stack.com/live-lambda-development) environment that allows you to work on your serverless apps live.
+
+```bash
+$ npx sst start
+```
+
+The first time you run this command it'll take a couple of minutes to deploy your app and a debug stack to power the Live Lambda Development environment.
+
+```
+===============
+ Deploying app
+===============
+
+Preparing your SST app
+Transpiling source
+Linting source
+Deploying stacks
+dev-react-app-my-stack: deploying...
+
+ ✅  dev-react-app-my-stack
+
+
+Stack dev-react-app-my-stack
+  Status: deployed
+  Outputs:
+    ApiEndpoint: https://51q98mf39e.execute-api.us-east-1.amazonaws.com
+```
+
+The `ApiEndpoint` is the API we just created. Let's test our endpoint using [insomnia](https://insomnia.rest/)
+
+![insomnia_post_request_with_correct_schema](/assets/examples/middy-validation-example/request1.png)
+
+As you can see the endpoint is working absolutely fine. We sent `mani` and `teja` as our `fname` and `lname` respectively and got `Hello, mani-teja` as the response
```

Comment:
I received `Hello, undefined-undefined.`. `event.body` is a string and `fname` and `lname` cannot be parsed.
<img width="1312" alt="Screen Shot 2021-11-10 at 6 03 22 PM" src="https://user-images.githubusercontent.com/83515/141207741-1cd419f8-275f-4db2-a31e-c075f2d6cd03.png">